lib/api.js

```diff
@@ -45,9 +34,11 @@ api.declare({
   title:       'Create New Secret',
   description: 'Set a secret associated with some key.'
 }, async function(req, res) {
-    req.satisfies(req.params);
     let {name} = req.params;
     let {secret, expires} = req.body;
+    if (!req.satisfies({name})) {
+      return;
```

Comment:
Oh, btw, this is a security fix... Forgetting to return and the function will proceed...

I guess we should make req.satisfies throw an exception and catch that in the API wrapper, that would be both backwards compatible and sensible... But for now we must return...

Note. it's possible to call req.satisfies in a way where it doesn't send a reply, but by default it does.... anyways, this only really a problem for the `set` method...